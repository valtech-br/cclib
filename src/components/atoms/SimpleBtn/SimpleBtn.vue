<template>
  <component
    :is="componentTag"
    v-bind="urlTag"
    class="simple-btn"
  >
    <span class="text-button noselect">
      <slot />
    </span>
  </component>
</template>

<script>
  export default {
    name: 'simpleBtn',
    props: {
      link: {
        type: [String, Object],
        default: ''
      }
    },
    computed: {
      isExternal() {
        if (!this.link) return false
        return (
          typeof this.link === 'string' && this.link.search(/(^\/|^#)/g) === -1
        )
      },
      isNative() {
        if (!this.link) return false
        return this.isExternal || !this.$router
      },
      urlTag() {
        if (!this.link) return ''
        return this.isNative ? { href: this.link, target: '_blank' } : { to: this.link }
      },
      componentTag() {
        if (!this.link) return 'div'
        const routerLink = this.$nuxt ? 'nuxt-link' : 'router-link'
        return this.isNative ? 'a' : routerLink
      }
    }
  }
</script>
