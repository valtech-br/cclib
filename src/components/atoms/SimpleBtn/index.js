import SimpleBtn from './SimpleBtn.vue'

export default SimpleBtn
