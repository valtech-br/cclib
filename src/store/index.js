import Vue from 'vue'
import Vuex, { Store } from 'vuex'
import global from './modules/global'

Vue.use(Vuex)

export default new Store({
  strict: true,
  modules: {
    global
  }
})
