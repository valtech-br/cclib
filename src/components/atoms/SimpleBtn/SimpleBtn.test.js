import { mount } from '@vue/test-utils'
import store from '@/store'
import SimpleBtn from './SimpleBtn.vue'

describe('Simple Button Component', () => {
  const Root = {
    template: `
      <simple-btn>
      </simple-btn>
    `
  }

  const context = {
    wrapper: null
  }

  beforeEach(() => {
    context.wrapper = mount(Root, {
      store,
      components: {
        [SimpleBtn.name]: SimpleBtn
      },
      attachTo: document.body
    })
  })

  it('Should renders correctly', () => {
    expect(context.wrapper.element).toMatchSnapshot()
  })
})
