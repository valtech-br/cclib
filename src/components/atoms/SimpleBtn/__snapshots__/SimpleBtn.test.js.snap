// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Button Component Should renders correctly 1`] = `
<div
  class="simple-btn"
>
  <span
    class="text-button noselect"
  />
</div>
`;
